### PR TITLE
fix(livekit): in-depth pass over reconnection, recovery and resiliency for LiveKit bridges

### PR DESCRIPTION
**Pull request by prlanzarin**
### What does this PR do?

This is an in-depth pass over LiveKit bridges' reconnection and recovery procedures.

The changes here stem from edge cases exploration over two source materials:
production data (logs/metrics) and internal testing.
The goal is to further harden resiliency of media when the client is off the
golden path.

Tests for this work are **not** in this PR. They live on a separate branch and
will follow as their own PR, because several of them are long-running and how we
want to run those in CI has to be discussed separately.

#### In detail

- **build(html5): livekit-client@2.19.0 (up from 2.17.3)** ([`5cb5d65b`](https://github.com/bigbluebutton/bigbluebutton/commit/5cb5d65b96ac5f67d335ad02a70843fdfd34368c))
- **fix(livekit): apply room options where livekit-client reads them** ([`075570e0`](https://github.com/bigbluebutton/bigbluebutton/commit/075570e0d3ef66f207e135c400cd3322c3a9d84b))
  - the SDK reads `stopLocalTrackOnUnpublish`, `dynacast` and `publishDefaults` from the constructor's options object by reference, and every later merge replaced it, so configured values never reached it.
- **fix(livekit): keep the primary Room object across membership changes** ([`5b4e9a62`](https://github.com/bigbluebutton/bigbluebutton/commit/5b4e9a622d929f426456f162be23e7a9802c3760))
  - the registry rebuilt the primary Room on re-acquire, so an eject and rejoin left the audio bridge publishing into a torn-down room until a reload.
- **fix(audio): do not unmute on a voice-conf leave under LiveKit** ([`f2528069`](https://github.com/bigbluebutton/bigbluebutton/commit/f252806996363f5755d307e6db55ac2820355404))
  - that rule assumes a FreeSWITCH breakout transfer; under LiveKit a participant leave is the ordinary reconnect, so a muted user came back publishing unmuted.
- **fix(livekit): skip an audio publish if already republished during a reconnect** ([`a7c0cd4f`](https://github.com/bigbluebutton/bigbluebutton/commit/a7c0cd4fb3c4c4ee44d409e804158b3a21d724cf))
  - the queued publish ran right after the SDK's own republish, and the duplicate timed out and was classified fatal, forcing a second reconnect.
- **fix(livekit): delay a server-triggered mute while the room reconnects** ([`a0947b26`](https://github.com/bigbluebutton/bigbluebutton/commit/a0947b26cf0e9301a3b94069cf2a938455f78b37))
  - the SFU reads a reconnect's transient unpublish as a mute, which silenced a user who never asked for it. The record is read once more when the room settles, so a mute dropped in error is not final.
- **fix(video): fail a LK camera publish cleanly when the preload is gone** ([`95247aa0`](https://github.com/bigbluebutton/bigbluebutton/commit/95247aa0e408b0fa993396c235417c8c5e0e811f))
  - nullish media streams threw uncaught on reconnects and detachments.
- **fix(audio): keep the local mute state set across a GraphQL disconnect** ([`0d6c7387`](https://github.com/bigbluebutton/bigbluebutton/commit/0d6c7387a2bd01d4a747c2a4ab1dab88b8e03bd2))
  - blanking the unmuted set made the button read muted while the microphone was still sending, and the audio state hooks acted on it.
- **fix(livekit): catch unpublish-after-mute failures** ([`651fa489`](https://github.com/bigbluebutton/bigbluebutton/commit/651fa4895a5025e6c67e0c64eb3dcf6ea4a5838b))
  - the deferred unpublish had no rejection handler; it is now cancelled when the room starts reconnecting.
- **fix(livekit): stale audio stream reference when doing a fallback publish** ([`004096bc`](https://github.com/bigbluebutton/bigbluebutton/commit/004096bc6ceb0ac2815a9b0fa9ce76b3880df829))
  - the fallback recorded the old, inactive stream rather than the capture it had just acquired.
- **fix(livekit): cleanup talking state if the room disconnects** ([`031e5930`](https://github.com/bigbluebutton/bigbluebutton/commit/031e5930dd93de3951805042513c80e15b8bfa6d))
  - the server-side cleanup has a deliberately long grace, which left talking indicators stuck.
- **fix(livekit): adjust reconn loop and notify users when it fails for good** ([`1afbbb45`](https://github.com/bigbluebutton/bigbluebutton/commit/1afbbb45adbb6c1b189bca003c4dec7b14ce584c))
  - self-hosted LiveKit has no built-in backoff, and an exhausted loop left a dead client with no indication of what happened.
- **fix(livekit): forcefully reconnect a stalled media session** ([`e16d0941`](https://github.com/bigbluebutton/bigbluebutton/commit/e16d094121f7a7e647e205add9aa3a37b3c4e40f))
  - recovery was conditioned on `RoomEvent.Disconnected`, which the SDK does not always emit, so the room never came back and said nothing. A room held out of connected past 60s is torn down and reconnected, keeping captures.
- **fix(video): surface unexpected camera teardowns to end users** ([`cdc59eb8`](https://github.com/bigbluebutton/bigbluebutton/commit/cdc59eb8f4271c21e2eedcdf2a9742cbf7233743))
  - device failures and reconnects raise a notice; lock settings and ejections get their own server-dispatched notifications, so a deliberate stop is not reported as a fault.
- **fix(livekit): republish a microphone that cannot carry audio** ([`fdfd9a74`](https://github.com/bigbluebutton/bigbluebutton/commit/fdfd9a74fc10cdd4392200f180003f915441ead8))
  - the SDK re-acquires a capture on unmute only for tracks it created itself, so an ended capture came back as an open microphone nobody could hear.
- **fix(livekit): keep a user's cameras in akka across a LiveKit reconnect** ([`bf74ac90`](https://github.com/bigbluebutton/bigbluebutton/commit/bf74ac90ff7cef0746532ac2cab733909cbcbcc5))
  - every forced reconnect closed and re-created the participant, and the handler stopped every webcam of that user. Cleanup moves to the BBB participant ejection. Requires the bbb-webrtc-sfu bump in this PR.
- **refactor(video): set livekit camera subscribe/unsubscribe logs to info** ([`b4493884`](https://github.com/bigbluebutton/bigbluebutton/commit/b44938845b0d68d0d74f79aede48292fdfb6b986))
- **refactor(livekit): pipe livekit-client's logs through the client logger** ([`8824c150`](https://github.com/bigbluebutton/bigbluebutton/commit/8824c150215b4ed304c2c11487f149a1321e018b))
  - SDK logs were console-only, so they could not be gathered from the field. Switchable via `public.media.livekit.sdkLogBridge`.
- **fix(video): resolve a camera against the room before subscribing** ([`5a10c6ef`](https://github.com/bigbluebutton/bigbluebutton/commit/5a10c6effdaa1c7f8d9ca4b79c53b34b0e752d1f))
  - a stale cache entry made the bridge ask for a sid the server no longer had, while the camera republished under a new sid stayed black.
- **feat(livekit): track subscriber offer-answer times on the client** ([`2d422a90`](https://github.com/bigbluebutton/bigbluebutton/commit/2d422a9076d65a6013e8b043b6b76b71370544c8))
  - the server logs only that an answer never arrived within 15s; this says whether the client was slow, never answered, or the offer was slow to reach it. Off by default.
- **fix(livekit): show the placeholder while a remote camera has video** ([`90fc95f3`](https://github.com/bigbluebutton/bigbluebutton/commit/90fc95f373596cd4e8fc094b96ad8f8957d109cb))
  - a dropped or unhealthy remote track left a black square instead of the user's placeholder.
- **build(bbb-webrtc-sfu): v2.25.0-beta.1 (up from v2.25.0-beta.0)** ([`520aaf38`](https://github.com/bigbluebutton/bigbluebutton/commit/520aaf382c3599f592438458e44bd5400ff4829f))
  - carries the deferred per-camera unpublish that `keep a user's cameras in akka across a LiveKit reconnect` depends on.

### Closes Issue(s)

Closes None

### How to test

There's extensive test coverage in a child branch based off this one that was used for the PR. Other than that, some manual hiccups worth remembering:

1. **Transparent signal reconnect**: kill the LiveKit websocket while unmuted.
   Media keeps flowing and the client must stay silent: no reconnect toast, and
   the mute button stays usable. A drop that lasts is still announced.
2. **Camera stopped by a moderator**: enable the Share Webcam lock, and
   separately use *Close cameras* on a sharing user. The user should be told
   what happened; neither should raise the "webcam interrupted" error.
3. **Stalled media**: black-hole the LiveKit signal socket without closing it.
   The client should give up after 60s, reconnect on its own, and say so.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/services/livekit/sdk-log-bridge.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  return match ? (match[1] as LogLevel) : undefined;
+};
+
+export const applyLiveKitSdkLogLevel = (level: LogLevel | string): void => {
+  const resolved = toLogLevel(level);
+
+  if (resolved === undefined) return;
+
+  setLogLevel(resolved);
+  setLogLevel(Math.min(resolved, LogLevel.info), LoggerNames.Engine);
```
[the remark this answers is not in the document]
Comment (prlanzarin) [reply bigbluebutton-html5/imports/ui/services/livekit/sdk-log-bridge.ts:104]: Read the commit log. This is intentional.

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/webcam/CameraHdlrHelpers.scala [hunk 2]
```diff
@@ -178,6 +184,17 @@ object CameraHdlrHelpers extends SystemConfiguration with RightsManagementTrait
...
+    val notifyEvent = MsgBuilder.buildNotifyUserInMeetingEvtMsg(
+      userId,
+      meetingId,
+      "info",
+      "video",
+      reason,
+      "Notification that a moderator or a lock setting stopped a user's camera",
+      Map("streamId" -> streamId)
+    )
+    outGW.send(notifyEvent)
```
[the remark this answers is not in the document]
Comment (prlanzarin) [reply akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/webcam/CameraHdlrHelpers.scala:196]: The quickest path will still be the notification one by definition (quickest path, least moving parts; akka-middleware-client vs akka-sfu-lk-akka-db-hasura-client. While a race may still exist, it's highly unlikely and addressing the race itself is something for later/a non issue for now.

File: bigbluebutton-html5/imports/ui/components/video-provider/livekit-camera-bridge/component.tsx [hunk 3]
```diff
@@ -500,20 +600,21 @@ const LiveKitCameraBridge: React.FC<LiveKitCameraBridgeProps> = ({
...
     return () => {
-      window.removeEventListener('beforeunload', exitVideo);
+      window.removeEventListener('beforeunload', handleBeforeUnload);
       evtRoom?.off(RoomEvent.TrackUnpublished, handleTrackUnpublished);
       evtRoom?.off(RoomEvent.TrackSubscribed, handleTrackSubscribed);
       evtRoom?.off(RoomEvent.TrackUnsubscribed, handleTrackUnsubscribed);
 
       VideoService.updatePeerDictionaryReference({});
-      exitVideo();
+      exitVideo(false);
```
[the remark this answers is not in the document]
Comment (prlanzarin) [reply bigbluebutton-html5/imports/ui/components/video-provider/livekit-camera-bridge/component.tsx:617]: This is wrong on all levels:
- stopStream's second parameter means `restarting`. Nothing restarts on a doc unmount.
- Every user initiated or server-recognized unshare marks the it as expected before the dock is flushed due to an empty video stream state - so it won't trigger the toast
- exitVideo's parameter defaults to expected, which is what a doc unmount means (previous item)
